From c58f1dd956df60be49b0ae151f1c209c9f6205b9 Mon Sep 17 00:00:00 2001
From: Maharaja Kennadyrajan <quic_mkenna@quicinc.com>
Date: Tue, 5 Nov 2024 16:41:26 +0530
Subject: [PATCH] UPSTREAM:CP: wifi: mac80211: Fix the BSSID for MLO STA

Use the associated AP link BSSID address to get the MLO STA info instead of
using the deflink BSSID in the function ieee80211_change_iface().

This fixes the upstream commit b65567b03 (wifi: mac80211: mlme: track
AP (MLD) address separately)

Patch-work: None
Patch-dependency: None

Signed-off-by: Maharaja Kennadyrajan <quic_mkenna@quicinc.com>
---
 net/mac80211/cfg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/mac80211/cfg.c b/net/mac80211/cfg.c
index 2f88f56..512764f 100644
--- a/net/mac80211/cfg.c
+++ b/net/mac80211/cfg.c
@@ -254,7 +254,7 @@ static int ieee80211_change_iface(struct wiphy *wiphy,
 			return 0;
 
 		mutex_lock(&local->sta_mtx);
-		sta = sta_info_get(sdata, sdata->deflink.u.mgd.bssid);
+		sta = sta_info_get(sdata, sdata->vif.cfg.ap_addr);
 		if (sta)
 			drv_sta_set_4addr(local, sdata, &sta->sta,
 					  params->use_4addr);
-- 
2.34.1

